libs/jsonforms-components/src/lib/Controls/FileUploader/FileUploaderControl.spec.tsx

```diff
@@ -0,0 +1,51 @@
+import { fireEvent, render } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import { ContextProvider, JsonFormContext as Context } from '../../Context/index';
+import { ControlElement, ControlProps } from '@jsonforms/core';
+import { FileUploader, FileUploaderLayoutRendererProps } from './FileUploaderControl';
+
+const fileUploaderUiSchema: ControlElement = {
+  type: 'Control',
+  scope: '#/properties/supportingDoc',
+  label: 'My First name',
+};
+
+const staticProps: FileUploaderLayoutRendererProps & ControlProps = {
+  uischema: fileUploaderUiSchema,
+  schema: {},
+  rootSchema: {},
+  handleChange: (path, value) => {},
+  enabled: true,
+  label: 'Supporting Document',
+  id: 'supportingDoc',
+  config: {},
+  path: '',
+  errors: '',
+  data: 'My file',
+  visible: true,
+  required: false,
+};
+
+const contextComponent = (props: FileUploaderLayoutRendererProps) => {
+  return (
+    <ContextProvider
+      fileManagement={{
+        fileList: {},
+        uploadFile: jest.fn(),
+        downloadFile: jest.fn(),
+        deleteFile: jest.fn(),
+      }}
+    >
+      <FileUploader {...props} /> ;
+    </ContextProvider>
+  );
+};
+
+describe('FileUploaderControl tests', () => {
+  it('can render fileupload control', () => {
+    const props = { ...staticProps };
+    const { container } = render(contextComponent(props));
```

Comment:
I ran into a bunch of trouble rendering like this, and it only got resolved when I changed it to:
render(<contextComponent {...props}/>);
It seems like it only rears its ugly head when there are "useState(...)" calls in the component, but its best to keep things consistent.

Also, the React standard is to capitalize the component name, so in this case you would call it "ContextComponent".

Reply:
ok  i fixed this and added one more unit test.